app/task_routes.py

```diff
@@ -0,0 +1,119 @@
+from flask import Blueprint, jsonify, abort, request 
+from app import db
+from app.models.task import Task
+from datetime import datetime
+import requests
+import os
+from dotenv import load_dotenv 
+
+load_dotenv()
+
+tasks_bp = Blueprint("task", __name__, url_prefix="/tasks")
+
+def valid_id(model, id):
+    """If ID is an int, returns the model object with that ID.
+        If ID is not an int, returns 400.
+        If model object with that ID doesn't exist, returns 404."""
+    try:
+        id = int(id)
+    except: 
+        abort(400, {"error": "invalid id"})
+    return model.query.get_or_404(id)
+
+@tasks_bp.route("", methods=["POST", "GET"])
+def handle_tasks(): 
+    """Handles post and get requests for tasks at /tasks"""
+    request_body = request.get_json()
+
+    if request.method == "POST": 
+        if "title" not in request_body or "description" not in request_body \
+        or "completed_at" not in request_body: 
+            return {"details": "Invalid data"}, 400
+
+        new_task = Task(title=request_body["title"],
+                    description=request_body["description"],
+                    completed_at=request_body["completed_at"])
+
+        db.session.add(new_task)
+        db.session.commit()
+
+        tasks_dict = {}
+        tasks_dict["task"] = new_task.to_dict()
+        return tasks_dict, 201
+
+    if request.method == "GET":
+        tasks_response = []
+        if request.args.get("sort") == "asc":
+            tasks = Task.query.order_by(Task.title)
+        elif request.args.get("sort") == "desc":
+            tasks = Task.query.order_by(Task.title.desc())
+        else: 
+            tasks = Task.query.all()
+        for task in tasks: 
+            tasks_response.append(task.to_dict())
```

Comment:
This definitely works well! Can you think of what this would look like as a list comprehension instead. Either is good, it's just a matter of personal preference~